src/cpp/session/modules/SessionHelp.R

```diff
@@ -676,39 +676,28 @@ options(help_type = "html")
                                     subset = TRUE,
                                     getSignature = FALSE)
 {
-   # Completions from the search path might have the 'package:' prefix, so
-   # lets strip that out.
-   package <- sub("package:", "", package, fixed = TRUE)
+   # Ensure topic and package are not zero-length
+   if (!length(package))
+      package <- ""
    
-   # Ensure topic is not zero-length
    if (!length(topic))
       topic <- ""
    
-   # If the topic is not provided, but we're getting help on e.g.
+   # Completions from the search path might have the 'package:' prefix, so
+   # lets strip that out.
+   package <- sub("package:", "", package, fixed = TRUE)
+   
+   # If the package is not provided, but we're getting help on e.g.
    # 'stats::rnorm', then split up the topic into the appropriate pieces.
-   if (!length(package) && any(grepl(":{2,3}", topic, perl = TRUE)))
+   if (package == "" && any(grepl(":{2,3}", topic, perl = TRUE)))
    {
       splat <- strsplit(topic, ":{2,3}", perl = TRUE)[[1]]
       topic <- splat[[2]]
       package <- splat[[1]]
    }
    
-   # If 'package' is the name of something on the search path, then we
-   # attempt to resolve the object and get its help.
-   if (length(package))
-   {
-      pos <- match(package, search(), nomatch = -1L)
-      if (pos >= 0)
-      {
-         object <- tryCatch(get(topic, pos = pos), error = function(e) NULL)
-         if (is.null(object))
-            return(NULL)
-         return(.rs.getHelpFromObject(object, envir, topic))
-      }
-   }
```

Comment:
Remove this bit because it is my understanding that things rather go the other way, e.g. `.rs.getHelpFromObject()` calls `.rs.getHelp()`. 

It should not change things because it was not called previously anyway as `package <- sub("package:", "", package, fixed = TRUE)` was done before.